### PR TITLE
Add dropout.

### PR DESCRIPTION
**Pull request by VariableVasasMT**
#### Description
Added dropout to ops and moved getScalar to ops of core as well. Will add another pull request to implement non default noiseShape.

Fixes https://github.com/tensorflow/tfjs/issues/117

---

##### For repository owners only:

Please remember to apply all applicable tags to your pull request.
Tags: FEATURE, BREAKING, BUG, PERF, DEV, DOC, SECURITY

For more info see: https://github.com/tensorflow/tfjs/blob/master/DEVELOPMENT.md

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-core/1343)

### CODE REVIEW

**Comment by nsthorat** [comment]
That would be really great, thanks :)

**Comment by nsthorat** [comment]
Let's just chat here, we don't have a public slack channel.

What do you mean by a global context? What do you want to do?

**Comment by VariableVasasMT** [comment]
@nsthorat seeded random number using mersenne from this https://en.wikipedia.org/wiki/Mersenne_Twister

**Comment by caisq** [comment]
@nsthorat PTAL?

File: src/ops/mersenne-rand.ts [hunk 1]
```diff
@@ -0,0 +1,129 @@
+/**
```
Comment (nsthorat) [inline src/ops/mersenne-rand.ts]: Hi sorry for the delay. Working with some open source folks at Google to okay this file. Just to check -- where is this implementation taken from?

Comment (nsthorat) [reply src/ops/mersenne-rand.ts]: So unfortunately we can't accept the copy paste of this code like this due to licensing issues -- is it possible to reimplement this algorithm using the source C++ simply as a reference?

**Comment by VariableVasasMT** [comment]
@nsthorat I used seedrandom's alea, I guess, I should have used it earlier, I am sorry for all the bullshit I took you through, it was very noob move, wont happen again, I went into a completely different direction which could have been avoided, please let me know if you need any corrections from me. Apologies!

**Comment by VariableVasasMT** [comment]
@dsmilkov please review

File: src/ops/mersenne-rand.ts (hunk 1 above)
Comment (VariableVasasMT) [reply src/ops/mersenne-rand.ts]: this file is deleted, but somehow still showing here, I think reviewable is displaying it deleted

**Comment by VariableVasasMT** [comment]
@dsmilkov please review

**Comment by dsmilkov** [comment]
One quick ask: the build fails since you added a new unit test and now have to re-run `yarn build` in order to update the auto-generated `src/tests.ts`. Once you commit the updated `src/tests.ts`, reply here and we can merge this!

See log: https://pantheon.corp.google.com/cloud-build/builds/26197cc6-bcc6-4139-978d-ce62447724f6?project=learnjs-174218

**Comment by VariableVasasMT** [comment]
> One quick ask: the build fails since you added a new unit test and now have to re-run `yarn build` in order to update the auto-generated `src/tests.ts`. Once you commit the updated `src/tests.ts`, reply here and we can merge this!
> 
> See log: https://pantheon.corp.google.com/cloud-build/builds/26197cc6-bcc6-4139-978d-ce62447724f6?project=learnjs-174218

@dsmilkov  fixed

**Pull request merged**